script.c

```diff
@@ -793,6 +793,30 @@ void script_print_vals(struct stack *stack) {
   putchar('\n');
 }
 
+void stack_print_hex(struct stack *stack) {
```

Comment:
I don't think this implementation is correct. This should show the encoded value of the stack value. For the case of empty arrays, this should be nothing!

```
	  for(i = 0; i < compiled_len; ++i)
		  printf("%02x", buf[i]);
```

Think about this case:

`btcs 0 ripemd160` -> `9c1185a5c5e9fc54612808977ee8f548b2258d31`
`btcs @00 ripemd160` -> `c81b94933420221a7ac004a90242d8b1d3e5070d`

we wouldn't want an empty array to serialize to something other than empty or else hash operations would be wrong.

Reply:
Ok I will roll back that change.